### ISSUE TITLE
Cancel filter button doesn't hide when value is ' '

### ISSUE THREAD

**Issue by nhduc296** [angular-ui/ng-grid#3824]
missing check string empty 
```
colFilter.term !== undefined && colFilter.term != null 
```
should be

```
colFilter.term !== undefined && colFilter.term != null && colFilter.term !== ''

**Comment by JLLeitschuh** [comment]
Not sure what issue this creates. I'm assuming you mean the select dropdown and not the string filter.
It seems to work fine here http://ui-grid.info/docs/#/tutorial/103_filtering

**Comment by nhduc296** [comment]
@JLLeitschuh I'm using UI-grid with angular 1.4.0

**Comment by JLLeitschuh** [comment]
Ui-Grid is currently not working with angular 1.4 #3658
We plan to support it soon.

**Comment by nhduc296** [comment]
It was fixed  in v3.0.0-rc.22.

https://github.com/angular-ui/ng-grid/commit/22e6acd81ece42f1133712199edc0142df920ed1
